decoder/src/elf2table/mod.rs

```diff
@@ -69,9 +94,18 @@ pub fn parse_impl(elf: &[u8], check_version: bool) -> Result<Option<Table>, anyh
     };
 
     if check_version {
-        self::check_version(version).map_err(anyhow::Error::msg)?;
+        self::check_version(&version).map_err(anyhow::Error::msg)?;
     }
 
+    let encoding = match encoding {
+        Some(e) => match &e[..] {
```

Comment:
nit: could you move this parsing into a `FromStr` implementation for `Encoding`?

Reply:
Went with `TryFrom<Option<String>>` in https://github.com/knurling-rs/defmt/pull/572